components/org.wso2.carbon.identity.query.saml/src/main/java/org/wso2/carbon/identity/query/saml/handler/SAMLAssertionFinderImpl.java

```diff
@@ -224,40 +209,46 @@ public Assertion findByID(String id) throws IdentitySAML2QueryException {
      * @throws  IdentitySAML2QueryException If unable to read assertions from database
      */
     private Assertion readAssertion(String id) throws IdentitySAML2QueryException {
-        String selectByIDquery = "SELECT SAML2_ASSERTION FROM IDN_SAML2_ASSERTION_STORE WHERE SAML2_ID =?";
+
         Connection connection = null;
         PreparedStatement preparedStatement = null;
         ResultSet resultSet = null;
+        Assertion assertion;
         try {
             connection = JDBCPersistenceManager.getInstance().getDBConnection();
-            preparedStatement = connection.prepareStatement(selectByIDquery);
+            preparedStatement = connection.prepareStatement(DBUtil.SELECT_BY_ID_QUERY);
             preparedStatement.setString(1, id);
             resultSet = preparedStatement.executeQuery();
-            if (resultSet.next()) {
-                String assertionString = resultSet.getString(1);
-                try {
-                    Assertion assertion = (Assertion) SAMLQueryRequestUtil.unmarshall(assertionString);
-                    log.debug("Assertion is retrieved from database with ID: " + id);
-                    return assertion;
-                } catch (IdentitySAML2QueryException e) {
-                    log.error("Unable to unmarshall assertions selected from database with ID: " + id, e);
-                    throw new IdentitySAML2QueryException(e.getMessage());
-                }
+            assertion = extractAssertions(resultSet).get(0);
+            if(assertion != null) {
```

Comment:
If we only have a debug level log in this block - we need to check whether debug is enabled - before this if condition.